searches/interpolation_search.py

```diff
@@ -48,14 +74,60 @@ def interpolation_search(sorted_collection, item):
     return None
 
 
-def interpolation_search_by_recursion(sorted_collection, item, left, right):
+def interpolation_search_by_recursion(
+    sorted_collection: list[int] | tuple[int], item: int, left: int, right: int
+) -> int | None:
     """Pure implementation of interpolation search algorithm in Python by recursion
     Be careful collection must be ascending sorted, otherwise result will be
     unpredictable
     First recursion should be started with left=0 and right=(len(sorted_collection)-1)
     :param sorted_collection: some ascending sorted collection with comparable items
     :param item: item value to search
     :return: index of found item or None if item is not found
+    >>> interpolation_search_by_recursion(
+    ... [-1, 0, 1], -1, 0, 2
+    ... )
```

Comment:
Please do not put tests on three lines unless your lines get to be longer than 88 characters.
```suggestion
    >>> interpolation_search_by_recursion([-1, 0, 1], -1, 0, 2)
```

Reply:
Got it! Will make a new commit. However, for the future, is there a way for a Python linter to be added as part of a pre commit check? I'm still new to doing open source contributions in Python but for Javascript projects I've worked on, it auto installed a linter when you did an ```npm install``` on the project.